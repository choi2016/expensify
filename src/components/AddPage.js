import React from 'react';

const ExpenseAddPage = () => (
    <div>
        This is my add component.
    </div>
); 

export default ExpenseAddPage;