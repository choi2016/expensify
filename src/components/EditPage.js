import React from 'react';

const ExpenseEditPage = (props) => {
    console.log(props);
    return (
        <div>
            Editing item number {props.match.params.id}.
        </div>
    );
}; 

export default ExpenseEditPage;