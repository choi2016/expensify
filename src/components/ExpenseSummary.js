import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpenseTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    const formatedCount = numeral(expenseCount).format("0,0");
    return (
        <div>
            <h3>Viewing {formatedCount} {expenseWord} totalling {formatedExpensesTotal}</h3>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpenseTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);