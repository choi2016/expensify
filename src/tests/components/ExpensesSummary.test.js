import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpenseSummary';

test('should render ExpensesSummary corectly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={55}/>)
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary corectly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={24} expensesTotal={55}/>)
    expect(wrapper).toMatchSnapshot();
});