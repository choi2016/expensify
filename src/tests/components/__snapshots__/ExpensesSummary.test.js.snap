// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary corectly with 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $0.55
  </h3>
</div>
`;

exports[`should render ExpensesSummary corectly with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    24
     
    expenses
     totalling 
    $0.55
  </h3>
</div>
`;
